product-mini/platforms/zephyr/simple-http/src/main.c

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (C) 2019 Intel Corporation.  All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ */
+
+// #include <autoconf.h>
+
+#include <stdlib.h>
+#include <string.h>
+#include "bh_platform.h"
+#include "bh_assert.h"
+#include "bh_log.h"
+#include "wasm_export.h"
+#include "http_get.h"
+
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/net/net_ip.h>
+#include <zephyr/net/socket.h>
+#include <zephyr/net/http/client.h>
+
+#define CONFIG_HEAP_MEM_POOL_SIZE WASM_GLOBAL_HEAP_SIZE
+#define CONFIG_APP_STACK_SIZE 8192
+#define CONFIG_APP_HEAP_SIZE 8192
+
+static char global_heap_buf[CONFIG_HEAP_MEM_POOL_SIZE] = { 0 };
+
+static int app_argc;
+static char **app_argv;
+
+int
+main(void)
+{
+    int start, end;
+    start = k_uptime_get_32();
+    uint8 *wasm_file_buf = NULL;
+    uint32 wasm_file_size;
+    wasm_module_t wasm_module = NULL;
+    wasm_module_inst_t wasm_module_inst = NULL;
+    RuntimeInitArgs init_args;
+    char error_buf[128];
+    const char *exception;
+
+    int log_verbose_level = 2;
+
+    memset(&init_args, 0, sizeof(RuntimeInitArgs));
+
+#if WASM_ENABLE_GLOBAL_HEAP_POOL != 0
+    init_args.mem_alloc_type = Alloc_With_Pool;
+    init_args.mem_alloc_option.pool.heap_buf = global_heap_buf;
+    init_args.mem_alloc_option.pool.heap_size = sizeof(global_heap_buf);
+    printf("global heap size: %d\n", sizeof(global_heap_buf));
+#else
+#error "memory allocation scheme is not defined."
+#endif
+
+    /* initialize runtime environment */
+    if (!wasm_runtime_full_init(&init_args)) {
+        printf("Init runtime environment failed.\n");
+        return;
+    }
+
+    bh_log_set_verbose_level(log_verbose_level);
+
+    /* load WASM byte buffer from byte buffer of include file */
+    wasm_file_buf = (uint8 *)wasm_test_file;
+    wasm_file_size = sizeof(wasm_test_file);
+    printf("Wasm file size: %d\n", wasm_file_size);
+
+    /* load WASM module */
+    if (!(wasm_module = wasm_runtime_load(wasm_file_buf, wasm_file_size,
+                                          error_buf, sizeof(error_buf)))) {
+        printf("Failed to load module: %s\n", error_buf);
+        goto fail1;
+    }
+
+    /* Set the WASI context */
+#if WASM_ENABLE_LIBC_WASI != 0
+#define ADDRESS_POOL_SIZE 1
+    const char *addr_pool[ADDRESS_POOL_SIZE] = {
+        "192.0.2.10/24",
```

Comment:
The fact that there is an address pool that acts as a white list should be mentioned in the Readme. Like a section `Customize the example to your environment` that lists all places that need to be touched if the board's and the server's IP need to be modified

Reply:
added a configuration section in readme.